build-logic/convention/src/main/kotlin/com/autonomousapps/convention/BaseConventionPlugin.kt

```diff
@@ -76,20 +76,34 @@ internal class BaseConventionPlugin(private val project: Project) {
       .orElse("false")
       .map { it.toBoolean() }
 
+    val taskGraph = gradle.taskGraph
+    val isFunctionalTest: Provider<Boolean> = providers
+      .provider { taskGraph.hasTask(":functionalTest") }
+
     tasks.withType(Sign::class.java).configureEach { t ->
       with(t) {
         inputs.property("version", publishedVersion)
         inputs.property("is-ci", isCi)
+        inputs.property("is-functional-test", isFunctionalTest)
 
         // Don't sign snapshots
         onlyIf("Not a snapshot") { !isSnapshot.get() }
         // We currently don't support publishing from CI
         onlyIf("release environment") { !isCi.get() }
+        // Don't sign when running functional tests
+        onlyIf("not running functional tests") { !isFunctionalTest.get() }
```

Comment:
Ideally this should be a little different: it should still sign if we're testing as part of the same flow as publishing to maven central. But the releasing notes already say to use `-x :functionalTest`, so I think it's not an issue in practice. Can return to this later if necessary.